File: FE/src/Model/ReportTypeVM.js
// src/models/ReportVM.js

import ReportVM from './ReportVM';
import BaseModel from './BaseModelVM';
export default class ReportTypeVM extends BaseModel {
  constructor(data = {}) {
    super(data.id || null, data.createdById, data.createdOn, data.modifiedById, data.modifiedOn, data.isDeleted);
    this.id = data.id || 0;
    this.name = data.name || '';
    this.reportPoint = data.ReportPoint || 0;
    this.reports = (data.reports || []).map(report => new ReportVM(report));
  }
}

File: BE/Models/ViewModels/ReportTypeVM.cs
namespace GoWheels_WebAPI.Models.ViewModels
{
    public class ReportTypeVM : BaseModelVM
    {
        public required string Name { get; set; }
        public required int ReportPoint { get; set; }
        public ICollection<ReportVM> ReportVMs { get; set; } = new List<ReportVM>();     
    }
}
